src/main/core/scheduler/pools/unbounded.rs

```diff
@@ -0,0 +1,474 @@
+use std::marker::PhantomData;
+use std::ops::Deref;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+
+use atomic_refcell::AtomicRefCell;
+
+use crate::utility::synchronization::count_down_latch::{
+    build_count_down_latch, LatchCounter, LatchWaiter,
+};
+use crate::utility::synchronization::semaphore::LibcSemaphore;
+
+// If making substantial changes to this scheduler, you should uncomment each test at the end of
```

Comment:
Optional - we're now running doc tests, and doc tests support ensuring that a snippet *won't* compile. See `compile_fail` in https://doc.rust-lang.org/rustdoc/write-documentation/documentation-tests.html

Reply:
I'll take a look at that, but I'm also considering trying https://github.com/dtolnay/trybuild, which allows us to ensure that not only it fails to compile, but also that it fails to compile for the correct reason. For example, I want these tests to fail due to lifetime issues rather than a syntax error or because we renamed something.

Reply:
ah cool - agreed that's a weakness of the `compile_fail` doc tests

Reply:
I'm going to address this in a follow-up PR since I'll also want to change the thread-per-host scheduler at the same time.